src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserHelperClasses.cs

```diff
@@ -459,7 +459,7 @@ public byte Updatability
 
         public bool IsReadOnly
         {
-            get => flags.HasFlag(_SqlMetadataFlags.IsReadOnlyMask);
+            get => (flags & _SqlMetadataFlags.IsReadOnlyMask) == 0;
```

Comment:
I dont think this is the right fix. 
I think _SqlMetadataFlags.IsReadOnlyMask needs to be modified instead of simply checking if this equates to 0.

Reply:
I think the `IsReadOnlyMask` enum name is a little misleading. Currently, 
```IsReadOnlyMask = (Updatable | UpdateableUnknown)``` (see [here](https://github.com/dotnet/SqlClient/blob/master/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserHelperClasses.cs#L408))
I think it should be named this instead:
```UpdatableBits = (Updatable | UpdateableUnknown)```
And in the Updatability getting, it'll look like this:
```get => (byte)(flags & _SqlMetadataFlags.UpdatableBits);```
and in the ReadOnly getter it'll look like this:
```get => (flags & _SqlMetadataFlags.UpdatableBits) == 0;``` meaning none of the updatable bit was on.

Reply:
I agree that IsReadOnlyMask is the wrong name for this flag.

I think `IsReadOnlyMask` should be changed to `IsUpdatableMask` I think the fix should say
`!flags.HasFlag(_SqlMetadataFlags.IsUpdatableMask);` 

What do you think ? 

Edit: I agree, we will have to do an & to check, !flag.HasFlag() wont work.

Reply:
@v-kaywon Can you change the name of IsReadOnlyMask  to IsUpdatableMask along with your change and update the PR? I agree with your fix. The name IsReadOnlyMask threw me off.